File: src/Bakabase.InsideWorld.Business/Components/Resource/Components/PropertyMatcher/MatchResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Bakabase.InsideWorld.Models.Constants;

namespace Bakabase.InsideWorld.Business.Components.Resource.Components.PropertyMatcher
{
    public class MatchResult
    {
        public MatchResultType Type { get; set; }
        public int? Layer { get; set; }
        public int? Index { get; set; }
        public string[]? Matches { get; set; }

        public static MatchResult OfLayer(int layer, int? index)
        {
            return new MatchResult
            {
                Type = MatchResultType.Layer,
                Layer = layer,
                Index = index
            };
        }

        public static MatchResult OfRegex(string[] matches)
        {
            return new MatchResult
            {
                Type = MatchResultType.Regex,
                Matches = matches
            };
        }
    }
}

File: src/Bakabase.InsideWorld.App.Core/ClientApp/src/components/PathSegmentsConfiguration/models/MatchResult.ts
import { MatchResultType } from '@/sdk/constants';

class MatchResult {
  type: MatchResultType;
  layer?: number;
  index?: number;
  matches?: string[];

  constructor(init?: Partial<MatchResult>) {
    Object.assign(this, init);
  }

  static OfLayer(layer: number, index?: number): MatchResult {
    return new MatchResult({
      type: MatchResultType.Layer,
      layer,
      index,
    });
  }

  static OfRegex(matches: string[]): MatchResult {
    return new MatchResult({
      type: MatchResultType.Regex,
      matches,
    });
  }
}

export {
  MatchResultType,
  MatchResult,
};
